api/v1alpha1/toolchainconfig_types.go

```diff
@@ -469,6 +477,36 @@ type TiersConfig struct {
 	TemplateUpdateRequestMaxPoolSize *int `json:"templateUpdateRequestMaxPoolSize,omitempty"`
 }
 
+// FeatureToggle defines a feature toggle/flag. Each feature is supposed to have a unique name.
+// Features are represented by kube object manifests in space and user templates.
+// Such manifests must have an annotation which refers to the corresponding feature name.
+// For example a manifest for a RoleBinding object in a space tier template with the following annotation:
+// "toolchain.dev.openshift.com/feature: os-lightspeed" would refer to a feature with "os-lightspeed" name.
+// When that template is applied for a new space then that RoleBinding object would be applied conditionally,
+// according to its weight.
+// +k8s:openapi-gen=true
+type FeatureToggle struct {
+	// A unique name of the feature
+	Name string `json:"name"`
+	// Rollout weight of the feature. An integer between 0-100.
+	// Any number lower than 0 will be treated as 0. Any number higher than 100 will be treated as 100.
+	// If not set then 100 is used by default.
+	// 0 means the corresponding feature should not be enabled at all, which means
+	// that corresponding template objects should not be applied at all.
+	// 100 means the feature should be always enabled (the template is always applied).
+	// The features are weighted independently of each other.
+	// For example if there are two features:
+	// - feature1, weight=5
+	// - feature2, weight=90
+	// And tiers (one or many) contain the following object manifests:
+	// - RoleBinding with "toolchain.dev.openshift.com/feature: feature1" annotation
+	// - ConfigMap with "toolchain.dev.openshift.com/feature: feature2" annotation
+	// Then the RoleBinding will be created for the corresponding tiers with probability of 0.05 (around 5 out of every 100 spaces would have it)
+	// And the ConfigMap will be created with probability of 0.9 (around 90 out of every 100 spaces would have it)
+	// +optional
+	Weight *int `json:"weight,omitempty"`
```

Comment:
it looks like it's a percentage value, another option would be to name this `WeightPercentage`

Reply:
yup, agree with @mfrancisc 
let's also add kubebuilder validation markers plus you can use `uint` (instead of `int`) to exactly match the usage 
```suggestion
	// +kubebuilder:validation:Minimum=0
	// +kubebuilder:validation:Maximum=100
	WeightPercentage *uint `json:"weightPercentage,omitempty"`
```

Reply:
I didn't want to use percentages because it's not really %
I'm concerned about misleading here. We do not guarantee that % of users will get the feature. We don't take into account how many users already have this feature for example. It's rather a probability of winning the "lottery" multiplied by 100. This is why I just call it weight which is genetic enough.

Reply:
But good point about the validation and the uint :+1:

Reply:
yeah, good point, I misread the description, sorry about the confusion and thanks for the clarification.

Reply:
Ok, I see, it's a _probability_ of getting the feature, not a percentage, I misunderstood the comment. thanks for the clarification 👍